client/packages/host/src/components/Initialise/hooks.ts

```diff
@@ -150,6 +155,7 @@ export const useInitialiseForm = () => {
     if (!syncStatus) return;
     // Need to be able to retry is syncStatus is erroneous
     setIsLoading(!syncStatus.error);
+    releaseWakeLock();
   }, [syncStatus]);
```

Comment:
Do you know in how far it is allowed to call acquireWakeLock or releaseWakeLock multiple time?

Anyway, I find it hard to check that every aquireWakeLock is released exactly once. For this kind of locking I would always put the `release` close to the `acquire` call. For example,

```js 
try {
  acquireWakeLock();
  await manualSync();
} finally {
  releaseWakeLock();
}
```

I think this is always possible here? So you only lock exactly for the time you do the sync.

Reply:
no, unfortunately. The `manualSync` call only initiates the sync, it doesn't wait for it to complete. We have to check for status, which is why the release is removed from the acquire. Agree though, ideally it would be in a `finally` so that we always release!

Reply:
fair enough, think @andreievg pointed out some potential race conditions, IMHO its an uphill battle to find them all (and keep it like this when there are future changes :-) ). Think would become much simpler if we could make a `sync(): Promise<>` that triggers the sync and waits for the sync status to arrive... don't know how easily this would be possible though. Maybe a helper hook where all the lock  and sync logic is encapsulated? so that the code is at least much easier to reason about?  e.g. this pseudo code:

```ts
const useSync = () => {
  const [isSyncing, setIsSyncing] = useState(false);

  const status = useSyncStatus();
  useEffect(() => {
    if (status === 'done') {
      if (isSyncing) {
        releaseLock();
        setIsSyncing(false);
      }
    }
  }, [status]);

  // cleanup when leaving the sync screen
  useEffect(() => {
    return () => {
      if (isSyncing) {
        releaseLock();
      }
    };
  }, []);

  // return a sync trigger
  return () => {
    if (isSyncing) {
      return;
    }
    acquireLock();
    setIsSyncing(true);
    manualSync();
  };
};
```